app/models/true_layer_bank.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class TrueLayerBank < ApplicationRecord
+  serialize :banks, Array
+
+  scope :by_updated_at, -> { order(updated_at: :asc) }
+  before_validation :populate_banks
+
+  validates :banks, presence: true
+
+  MOCK_BANK = {
+    provider_id: 'mock',
+    display_name: 'Mock Bank',
+    logo_url: 'https://truelayer-client-logos.s3-eu-west-1.amazonaws.com/banks/banks-icons/mock-icon.svg'
+  }.freeze
+
+  def self.banks
+    TrueLayerBanksUpdateWorker.perform_in 10.seconds
+    instance = by_updated_at.last || create!
+    ([mock_bank] + instance.banks).compact
```

Comment:
Really minor: I think the only situation where there will be a `nil` in the array is if `mock_bank` is nil. By using `compact` that call operates over the whole array every time. Would this be better:
```
return instance.banks unless mock_bank

[mock_bank] + instance.banks
```

Reply:
That's much better, thanks